com.unity.sg2/Editor/GraphUI/DataModel/GraphDataNodeModel.cs

```diff
@@ -256,6 +256,24 @@ public void OnPreviewShaderCompiling()
             PreviewShaderIsCompiling = true;
         }
 
+        public override void OnConnection(IPortModel selfConnectedPortModel, IPortModel otherConnectedPortModel)
+        {
+            base.OnConnection(selfConnectedPortModel, otherConnectedPortModel);
+            if (selfConnectedPortModel.Direction == PortDirection.Input)
+            {
+                DefineNode(); // TODO: This doesn't seem to work.
+            }
+        }
+
+        public override void OnDisconnection(IPortModel selfConnectedPortModel, IPortModel otherConnectedPortModel)
+        {
+            base.OnDisconnection(selfConnectedPortModel, otherConnectedPortModel);
+            if (selfConnectedPortModel.Direction == PortDirection.Input)
+            {
+                DefineNode();
+            }
+        }
```

Comment:
The above results in correct behavior in a few cases, but the UI doesn't update reliably from doing this.